### DISCUSSION TITLE
Container tag/name prefix

### DISCUSSION THREAD

**Discussion by AdamJessop** [coollabsio/coolify#3569] [Feature Requests]
It would be nice to prefix containers with a tag so we can identify containers without losing the rollback functionality.

We use datadog across our coolify servers and it would help with identifying containers and what they are serving.

I am aware we can force name the containers, but at the cost of rolling deployments etc. Instead it would be great if we could supply a prefix and a random string is postfixed to the container name to keep the rolling deployment feature intact.

**Comment by Julien-R44** [comment]
Hey, I'm experiencing the exact same issue and just wanted to explain my use case

I have a Grafana dashboard along with cAdvisor running on my server where Coolify is set up. cAdvisor exports CPU/RAM usage metrics for each container individually. Here’s a quick screenshot:

![image](https://github.com/user-attachments/assets/4370c15d-0aa8-410d-985d-7a02da4bf1ba)

As you can see, it's impossible to identify anything with these auto-generated container names. So yeah, I was thinking about 2 solutions:

- As suggested by the OP, implementing a prefix system for container names.
- Additionally, I think it would be super cool [to automatically add](https://github.com/coollabsio/coolify/blob/9af3fe9c973b1a23f1feb183c3a89213b2e37f0c/bootstrap/helpers/docker.php#L190) `coolify.projectName` and `coolify.applicationName` labels to each container. The values for these labels would be automatically pulled from previously entered form inputs. So, for example, this application:
	![image](https://github.com/user-attachments/assets/a0e9a47f-5e93-4e1c-b240-389ac9f035a8)
	would have `coolify.projectName=myproject` and `coolify.applicationName=my-super-project`.

I think both solutions would be useful to have. In fact, the second one might be more practical in some cases: it would allow me to get consistent metrics for a single container, even if it restarts and gets a new name. With the labels, the values would remain the same even if it restart, allowing me to maintain a continuous series instead of having two separate series labeled `my-prefix-3422394823948234` and `my-prefix-3249238429384`. Hope that makes sense 😅 !

I would be happy to submit a PR if we get the go-ahead for this.

Also, on top of that, we also have an issue with managed DBs, where we can't edit container names at all. Unfortunately I don't know enough about Coolify's inner workings to suggest a solution here.

**Reply by fma965** [reply]
Would love to see this, another thing that may be useful in relation to this is to ensure the traefik/caddy labels are named the correct services based on the docker containers.

**Comment by andrasbacsai** [comment]
https://github.com/coollabsio/coolify/pull/4324 has been merged, so new labels will be added in the next version of Coolify (ofc your need to redeploy your resource to get the new labels).

**Reply by Cinzya** [reply]
This is currently not solving everyone's problem, as this only adds new container labels such as `coolify.resourceName`.

OP was however specifically looking for a prefix container name feature, as some apps only read from the container name.

**Comment by gzucca** [comment]
I was looking for this exact thing too. I'm using Loki with FluentBit logs, but I only have the "container_name" to label the logs, and a preffix for each container would solve all my problems.

**Comment by peturgeorgievv** [comment]
Just to note, with aliases you can now setup a full integration with Graphana, Prometheus and cAdvisor and see the alias on the dashboard, so its' easily visible. I'll write an article at some point when i have to the time, but with ChatGPT and some twaeking it works amazingly well.

**Reply by karl-barbour** [reply]
Hi @peturgeorgievv - could you possibly explain this more, please?

**Reply by peturgeorgievv** [reply]
I don't have the time to write the article still, probably end of next month... In short:

- cAdvisor as docker-compose - 'gcr.io/cadvisor/cadvisor:latest' in the server you want metrics from
- Prometheus. I have one repository with simple Dockerfile, and prometheus configuration. This below is the important part as target_label:

`  # cAdvisor on each host (captures container metrics + network‑alias as "alias")
  - job_name: 'cadvisor'
    metrics_path: /metrics
    static_configs:
      - targets:
          - 'cadvisor-sgks8gk4oggsk8ccokwcwoc4:9109'
          - '10.255.0.4:9110'
        labels:
          service: 'cadvisor'
    metric_relabel_configs:
      # 1) fallback: use Docker 'name' label for alias
      - source_labels: [name]
        regex: (.+)
        target_label: alias

      # 2) if Coolify injected serviceName, overwrite alias
      - source_labels: [container_label_coolify_serviceName]
        regex: (.+)
        target_label: alias

      # 3) if you’ve added your own 'alias=' Docker label, that wins
      - source_labels: [container_label_alias]
        regex: (.+)
        target_label: alias

      # drop any metrics without an alias
      - source_labels: [alias]
        regex: ""
        action: drop`
       
I do have some psecific stuff too though there, but with some AI you'll figure it out for your setup. 

The next thing is in Grafana to connect Prometheus as Data Source, and explain all i told you to ChatGPT so it gives you a proper Dashboard.

**Comment by hhannis** [comment]
Would say this is critical. It causes problem many places particular when you run several containers with same image. Ie having 10 pocketbase , 10 wordpress etc.

You get problems when:
1. You want to see docker memory with "docker stats" or "bezsel and similar
2. You want to run terminal in coolify UI and the dropdown in terminal contains 10 pocketbase-RANDOM GUID
3. You want to upload hooks into pocketbase, and the docker volume folder contains the same to pocketbase-RANDOM GUID folders

This is just the problems I run into with less than 24 hours experience with Coolify.... Must be fixed.

**Comment by Bumpy226** [comment]
In case people are still running into this issue, Coolify checks for the following environment variables and appends them to the log drains before sending them:

`COOLIFY_APP_NAME`
`COOLIFY_PROJECT_NAME`
`COOLIFY_SERVER_IP`
`COOLIFY_ENVIRONMENT_NAME`

Setting these manually in your project makes them appear in the log drains, allowing you to filter on them in Grafena, Axiom, etc. while still allowing for rolling updates with randomized container names.

**Comment by peturgeorgievv** [comment]
As people we asking a lot of times about the grafana stuff, here is the article I promised to write. It has fully working examples and JSON for grafana. https://peturgeorgievv.com/blog/deploy-prometheus-and-grafana-in-coolify-code-example
